### PR TITLE
 Create graphs/dijkstra_alternate.py

### PR DESCRIPTION
**Pull request by AtulRajput01**
### Describe your change:

* [x] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: graphs/dijkstra.py [hunk 1]
```diff
@@ -1,118 +1,58 @@
...
+    [0, 4, 0, 0, 0, 0, 0, 8, 0],
+    [4, 0, 8, 0, 0, 0, 0, 11, 0],
+    [0, 8, 0, 7, 0, 4, 0, 0, 2],
+    [0, 0, 7, 0, 9, 14, 0, 0, 0],
+    [0, 0, 0, 9, 0, 10, 0, 0, 0],
+    [0, 0, 4, 14, 10, 0, 2, 0, 0],
+    [0, 0, 0, 0, 0, 2, 0, 1, 6],
+    [8, 11, 0, 0, 0, 0, 1, 0, 7],
+    [0, 0, 2, 0, 0, 0, 6, 7, 0],
+]
```
Comment (AtulRajput01) [inline graphs/dijkstra.py:56]: i think this isuue come due to syntax of line 56

**Comment by ZeroDayOwl** [comment]
@AtulRajput01 check pre-commit details and fix the issues.

File: graphs/dijkstra.py [hunk 2]
```diff
@@ -1,118 +1,58 @@
...
-    """Return the cost of the shortest path between vertices start and end.
+    def dijkstra(self, src):
 
-    >>> dijkstra(G, "E", "C")
-    6
-    >>> dijkstra(G2, "E", "F")
-    3
-    >>> dijkstra(G3, "E", "F")
-    3
-    """
```
Comment (ZeroDayOwl) [inline graphs/dijkstra.py:45]: Why did you remove the doctest?

File: graphs/dijkstra.py [hunk 3]
```diff
@@ -1,118 +1,58 @@
...
-    -------- 2 ---------> G ------- 1 ------
-"""
-G3 = {
-    "B": [["C", 1]],
-    "C": [["D", 1]],
-    "D": [["F", 1]],
-    "E": [["B", 1], ["G", 2]],
-    "F": [],
-    "G": [["F", 1]],
-}
```
Comment (ZeroDayOwl) [inline graphs/dijkstra.py:104]: Why did you remove this part as well?

**Review by ZeroDayOwl — CHANGES_REQUESTED** [review]

**Comment by cclauss** [comment]
Instead of deleting someone else's work, let's move this to a separate file `graphs/dijkstra_alternate.py` and make sure that naming, tests, type hints are in compliance with https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md

**Review by AtulRajput01 — COMMENTED** [review]
code fix

**Comment by AtulRajput01** [comment]
> @AtulRajput01 check pre-commit details and fix the issues.

code fix .

**Comment by cclauss** [comment]
graphs/dijkstra.py:6:10: N802 function name 'printSolution' should be lowercase
graphs/dijkstra.py:11:10: N802 function name 'minDistance' should be lowercase
graphs/dijkstra.py:11:34: N803 argument name 'sptSet' should be lowercase
graphs/dijkstra.py:13:9: A001 variable "min" is shadowing a python builtin
graphs/dijkstra.py:16:44: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
graphs/dijkstra.py:17:17: A001 variable "min" is shadowing a python builtin
graphs/dijkstra.py:26:10: N806 variable 'sptSet' in function should be lowercase
graphs/dijkstra.py:28:13: B007 Loop control variable 'cout' not used within the loop body. If this is intended, start the name with an underscore.
graphs/dijkstra.py:36:35: E712 comparison to False should be 'if cond is False:' or 'if not cond:'

**Comment by AtulRajput01** [comment]
> graphs/dijkstra.py:6:10: N802 function name 'printSolution' should be lowercase graphs/dijkstra.py:11:10: N802 function name 'minDistance' should be lowercase graphs/dijkstra.py:11:34: N803 argument name 'sptSet' should be lowercase graphs/dijkstra.py:13:9: A001 variable "min" is shadowing a python builtin graphs/dijkstra.py:16:44: E712 comparison to False should be 'if cond is False:' or 'if not cond:' graphs/dijkstra.py:17:17: A001 variable "min" is shadowing a python builtin graphs/dijkstra.py:26:10: N806 variable 'sptSet' in function should be lowercase graphs/dijkstra.py:28:13: B007 Loop control variable 'cout' not used within the loop body. If this is intended, start the name with an underscore. graphs/dijkstra.py:36:35: E712 comparison to False should be 'if cond is False:' or 'if not cond:'

done sir

File: graphs/dijkstra.py [hunk 4]
```diff
@@ -3,18 +3,18 @@ def __init__(self, vertices):
         self.V = vertices
         self.graph = [[0 for column in range(vertices)] for row in range(vertices)]
 
-    def printSolution(self, dist):
+    def printsolution(self, dist):
```
Comment (cclauss) [inline graphs/dijkstra.py:6]:
```suggestion
    def print_solution(self, distances: list[int]):
```

File: graphs/dijkstra.py [hunk 5]
```diff
@@ -3,18 +3,18 @@ def __init__(self, vertices):
         self.V = vertices
         self.graph = [[0 for column in range(vertices)] for row in range(vertices)]
```
Comment (cclauss) [inline graphs/dijkstra.py:4]:
```suggestion
        self.graph = [[0] * vertices for row in range(vertices)]
```

File: graphs/dijkstra.py [hunk 6]
```diff
@@ -3,18 +3,18 @@ def __init__(self, vertices):
...
 
-    def printSolution(self, dist):
+    def printsolution(self, dist):
         print("Vertex \t Distance from Source")
         for node in range(self.V):
             print(node, "\t\t", dist[node])
 
-    def minDistance(self, dist, sptSet):
+    def mindistance(self, dist, sptSet):
```
Comment (cclauss) [inline graphs/dijkstra.py:11]:
```suggestion
    def minimum_distance(self, distances, spt_set) -> int:
```
What is `spt_set` supposed to be?!?  There must be a better name.

**Review by cclauss — CHANGES_REQUESTED** [review]
According to CONTRBUTING.md there needs to be at least one algorithmic function that takes in one or more parameters and returns one or more values.  Algorithmic functions should not print().

**Comment by AtulRajput01** [comment]
> According to CONTRBUTING.md there needs to be at least one algorithmic function that takes in one or more parameters and returns one or more values. Algorithmic functions should not print().


okay sir ,i understood very well.

**Review by Abinashbunty — CHANGES_REQUESTED** [review]
- In line 30, during function call, it should be `self.minimum_distance`
- All `dist` should be `distances` or vice versa.
- All `sptset` should be `spt_set` or vice versa.
- `printsolution` should be `print_solution` or vice versa.
- Include doctest.
- Provide some comments in algorithm execution.

**Comment by Abinashbunty** [comment]
A general advice is to execute the code online/locally to fix the compilation(interpreter) errors. 👍🏻

File: graphs/dijkstra.py (hunk 1 above)
Comment (AtulRajput01) [reply graphs/dijkstra.py:75]: okay now it's working

**Comment by AtulRajput01** [comment]
> A general advice is to execute the code online/locally to fix the compilation(interpreter) errors. 👍🏻

it's working on my machine as well.

**Comment by cclauss** [comment]
graphs/dijkstra.py:22:48: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
graphs/dijkstra.py:37:13: B007 Loop control variable 'cout' not used within the loop body. If this is intended, start the name with an underscore.
graphs/dijkstra.py:55:35: E712 comparison to False should be 'if cond is False:' or 'if not cond:'

File: graphs/dijkstra.py [hunk 7]
```diff
@@ -1,118 +1,77 @@
...
+    # not yet included in shortest path tree
+    def mindistance(self, dist, sptset):
+
+        # Initialize minimum distance for next node
+        minimum = 1e7
+
+        # Search not nearest vertex not in the
+        # shortest path tree
+        for v in range(self.V):
+            if dist[v] < minimum and sptset[v] == False:
```
Comment (cclauss) [inline graphs/dijkstra.py:22]:
```suggestion
            if dist[v] < minimum and sptset[v] is False:
```

File: graphs/dijkstra.py [hunk 8]
```diff
@@ -1,118 +1,77 @@
...
+    # Function that implements Dijkstra's single source
+    # shortest path algorithm for a graph represented
+    # using adjacency matrix representation
+    def dijkstra(self, src):
+
+        dist = [1e7] * self.V
+        dist[src] = 0
+        sptset = [False] * self.V
+
+        for cout in range(self.V):
```
Comment (cclauss) [inline graphs/dijkstra.py:37]:
```suggestion
        for _ in range(self.V):
```

File: graphs/dijkstra.py [hunk 9]
```diff
@@ -1,118 +1,77 @@
...
+            sptset[u] = True
+
+            # Update dist value of the adjacent vertices
+            # of the picked vertex only if the current
+            # distance is greater than new distance and
+            # the vertex in not in the shortest path tree
+            for v in range(self.V):
+                if (
+                    self.graph[u][v] > 0
+                    and sptset[v] == False
```
Comment (cclauss) [inline graphs/dijkstra.py:48]:
```suggestion
                    and sptset[v] is False
```

File: graphs/dijkstra.py [hunk 10]
```diff
@@ -1,75 +1,77 @@
...
-	def mindistance(self, dist, sptset):
+        # Initialize minimum distance for next node
+        minimum = 1e7
 
-		# Initialize minimum distance for next node
-		minimum = 1e7
+        # Search not nearest vertex not in the
+        # shortest path tree
+        for v in range(self.V):
+            if dist[v] < minimum and sptset[v] == False:
```
Comment (Abinashbunty) [inline graphs/dijkstra.py:22]: Modify this to ⬇️ 

```console
if distances[v] < minimum and not spt_set[v]:
```

**Review by AtulRajput01 — COMMENTED** [review]
+1

File: graphs/dijkstra.py [hunk 11]
```diff
@@ -1,75 +1,77 @@
...
 
-			# Put the minimum distance vertex in the
-			# shortest path tree
-			sptset[u] = True
+            # Update dist value of the adjacent vertices
+            # of the picked vertex only if the current
+            # distance is greater than new distance and
+            # the vertex in not in the shortest path tree
+            for v in range(self.V):
+                if (
```
Comment (Abinashbunty) [inline graphs/dijkstra.py:46]: Change this if statement like the other comment above as well so that you won't have to equate anything to `False`. Avoid using `False`. Use `not` `<false condition>` instead.

**Review by Abinashbunty — CHANGES_REQUESTED** [review]
Moreover, check my comment above which Christian has edited. Use `distances`, `spt_set` and `print_solution`.

**Review by cclauss — CHANGES_REQUESTED** [review]

**Comment by cclauss** [comment]
OK.  I changed the filename, used self-documenting names for function and variable, added type hints, and doctests.

I still have no clue what a better name for `sptset` would be except perhaps `visited`.

**Comment by Abinashbunty** [comment]
@cclauss `visited` seems the only logical variable name for `sptset`.

File: graphs/dijkstra_alternate.py [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+    [0, 8, 0, 7, 0, 4, 0, 0, 2],
+    [0, 0, 7, 0, 9, 14, 0, 0, 0],
+    [0, 0, 0, 9, 0, 10, 0, 0, 0],
+    [0, 0, 4, 14, 10, 0, 2, 0, 0],
+    [0, 0, 0, 0, 0, 2, 0, 1, 6],
+    [8, 11, 0, 0, 0, 0, 1, 0, 7],
+    [0, 0, 2, 0, 0, 0, 6, 7, 0],
+]
+
+g.dijkstra(0)
```
Comment (cclauss) [inline graphs/dijkstra_alternate.py:99]:
```suggestion
if __name__ == "__main__":
    graph = Graph(9)
    graph.graph = [
        [0, 4, 0, 0, 0, 0, 0, 8, 0],
        [4, 0, 8, 0, 0, 0, 0, 11, 0],
        [0, 8, 0, 7, 0, 4, 0, 0, 2],
        [0, 0, 7, 0, 9, 14, 0, 0, 0],
        [0, 0, 0, 9, 0, 10, 0, 0, 0],
        [0, 0, 4, 14, 10, 0, 2, 0, 0],
        [0, 0, 0, 0, 0, 2, 0, 1, 6],
        [8, 11, 0, 0, 0, 0, 1, 0, 7],
        [0, 0, 2, 0, 0, 0, 6, 7, 0],
    ]
    graph.dijkstra(0)
```

**Comment by ZeroDayOwl** [comment]
> > A general advice is to execute the code online/locally to fix the compilation(interpreter) errors. 👍🏻
> 
> it's working on my machine as well, then why pre-commit failure accur

Did you run the tests as mentioned in the contributing guideline?

**Comment by AtulRajput01** [comment]
ok

**Review by cclauss — APPROVED** [review]
